### PR TITLE
gcp - added patch deployment

### PR DESCRIPTION
**Pull request by evhenyshaliov**
Use case:

`policies:
  - name: os-patch-deployment-configured-with-recurring-schedule
    description: |
      OS Patch Deployment is not configured with a recurring schedule
    resource: gcp.patch-deployment
    filters:
      - type: value
        key: recurringSchedule
        value: absent`

### CODE REVIEW

File: tools/c7n_gcp/tests/test_query.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def test_gcp_resource_metadata_asset_type():
         'ml-job',
         'ml-model',
         'notebook',
+        'patch-deployment',
```
Comment (kapilt) [inline tools/c7n_gcp/tests/test_query.py:37]: asset inventory supports patch deployment, add the corresponding metadata to the resource type info instead
https://cloud.google.com/asset-inventory/docs/supported-asset-types

Comment (evhenyshaliov) [reply tools/c7n_gcp/tests/test_query.py:37]: Fixed. Please check

File: tools/c7n_gcp/c7n_gcp/resources/osconfig.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    """ GC resource: https://cloud.google.com/compute/docs/osconfig/rest/v1/projects.patchDeployments"""
+    class resource_type(TypeInfo):
+        service = 'osconfig'
+        version = 'v1'
+        component = 'projects.patchDeployments'
+        enum_spec = ('list', 'patchDeployments[]', None)
+        scope_key = 'parent'
+        name = id = 'id'
+        scope_template = 'projects/{}'
+        default_report_fields = ['displayName', 'expireTime']
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/resources/osconfig.py:18]: invalid report fields

Comment (kapilt) [reply tools/c7n_gcp/c7n_gcp/resources/osconfig.py]: fixed.

File: tools/c7n_gcp/c7n_gcp/resources/osconfig.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+@resources.register('patch-deployment')
+class PatchDeployment(QueryResourceManager):
+    """ GC resource: https://cloud.google.com/compute/docs/osconfig/rest/v1/projects.patchDeployments"""
+    class resource_type(TypeInfo):
+        service = 'osconfig'
+        version = 'v1'
+        component = 'projects.patchDeployments'
+        enum_spec = ('list', 'patchDeployments[]', None)
+        scope_key = 'parent'
+        name = id = 'id'
```
[the remark this answers is not in the document]
Comment (kapilt) [reply tools/c7n_gcp/c7n_gcp/resources/osconfig.py]: fixed.

**Review by kapilt — APPROVED** [review]
lgtm

**Pull request merged**